### ISSUE TITLE
YouTube replacement messes up playlist embeds

### ISSUE THREAD

**Issue by fluffy-critter** [RomanSixty/Feed-on-Feeds#46]
If a feed embeds a YouTube playlist rather than a video, the player replacement thing screws up the embed.

Example feed with some recent content of those sorts: https://novembeat.com/feed

[original item](http://novembeat.com/works/79-Gero-Takke-Commodore-Amigo):

```
    <entry>
        <title>Works: Gero Takke / Commodore Amigo</title>
        <link href="https://novembeat.com/works/79-Gero-Takke-Commodore-Amigo" rel="alternate" type="text/html" />
        <published>2020-11-16T09:59:15-08:00</published>
        <updated>2020-11-16T09:59:15-08:00</updated>
        <id>urn:uuid:1a19bf35-9b59-58ea-80ff-05584896a490</id>
        <author><name>Gero Takke</name></author>
        <content type="html"><![CDATA[

<p><a href="https://www.youtube.com/playlist?list=PLXwJEk7hkqgdLX4PeBL9xAZ8V0CevVC3g">YouTube playlist</a>:</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/videoseries?list=PLXwJEk7hkqgdLX4PeBL9xAZ8V0CevVC3g" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>






        ]]></content>
        <category term="https://novembeat.com/" label="Novembeat" />
        <category term="https://novembeat.com/works/" label="Works" />
        
        

    </entry>
```

Replaced item:

```
<div class="body"><p><a href="https://www.youtube.com/playlist?list=PLXwJEk7hkqgdLX4PeBL9xAZ8V0CevVC3g" target="_blank">YouTube playlist</a>:</p>
<div class="youtube-video" data-ytid="videoseries" style="background-image: url('//i.ytimg.com/vi/videoseries/hqdefault.jpg')" onclick="embed_youtube(this);"></div></div>
```

After clicking on the image:

```
<div class="body"><p><a href="https://www.youtube.com/playlist?list=PLXwJEk7hkqgdLX4PeBL9xAZ8V0CevVC3g" target="_blank">YouTube playlist</a>:</p>
<iframe src="https://www.youtube.com/embed/videoseries?autoplay=1&amp;rel=0" allowfullscreen="1" width="560" height="315" frameborder="0"></iframe></div>
```

**Comment by RomanSixty** [comment]
Tried to fix this with d3e5036b1de5b39521bb47e40a2fe84b14ef0027

To also show a proper preview image one has to use the YouTube API which requires an API key (single video previews can be generated just by URL manipulation). I don't think getting and storing API keys for third party services would be a good idea for a feed reader so I left it in this "working but not particularly beautiful" state. I never came across feeds with YouTube playlists in their items so far anyway.

If this is nagging you too much you might just disable the YouTube plugin. Its main use is to make single video embeds a bit faster since it only loads the preview image, not the whole player for each item.

**Comment by fluffy-critter** [comment]
Yeah, that's pretty sensible, although maybe it'd make more sense to have the plugin skip playlists entirely? It's nice having the on-demand loader for single videos, but since playlists are pretty uncommon in RSS feeds I think it'd be okay for them to just be loaded normally.

**Comment by RomanSixty** [comment]
Ah well, why not the simple solution, fair point ;)

Fixed that in 4e37c1b7f20728467328be00266b0c71f5a0e748. There's an edge case in which an item might contain a playlist embed among single videos, in which case the whole item will not be treated by the plugin. But I don't think this will be a common case.
